From c8a950d0d3b926a02c7b2e713850d38217cec3d1 Mon Sep 17 00:00:00 2001
Message-Id: <c8a950d0d3b926a02c7b2e713850d38217cec3d1.1649665150.git.geliang.tang@suse.com>
From: Jean-Philippe Brucker <jean-philippe@linaro.org>
Date: Tue, 10 Nov 2020 17:43:05 +0100
Subject: [PATCH] tools: Factor HOSTCC, HOSTLD, HOSTAR definitions
Git-commit: c8a950d0d3b926a02c7b2e713850d38217cec3d1
Patch-mainline: v5.11-rc1
References: bsn#127
Modified-by-SEL: No


Several Makefiles in tools/ need to define the host toolchain variables.
Move their definition to tools/scripts/Makefile.include

Signed-off-by: Jean-Philippe Brucker <jean-philippe@linaro.org>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Acked-by: Jiri Olsa <jolsa@redhat.com>
Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Link: https://lore.kernel.org/bpf/20201110164310.2600671-2-jean-philippe@linaro.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/scripts/Makefile.include |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/tools/scripts/Makefile.include
+++ b/tools/scripts/Makefile.include
@@ -77,6 +77,16 @@ HOSTCC  ?= gcc
 HOSTLD  ?= ld
 endif
 
+ifneq ($(LLVM),)
+HOSTAR  ?= llvm-ar
+HOSTCC  ?= clang
+HOSTLD  ?= ld.lld
+else
+HOSTAR  ?= ar
+HOSTCC  ?= gcc
+HOSTLD  ?= ld
+endif
+
 ifeq ($(CC_NO_CLANG), 1)
 EXTRA_WARNINGS += -Wstrict-aliasing=3
 endif
